litestar/connection/websocket.py

```diff
@@ -72,7 +72,7 @@ def receive_wrapper(self, receive: Receive) -> Receive:
 
         async def wrapped_receive() -> ReceiveMessage:
             if self.connection_state == "disconnect":
-                raise WebSocketException(detail=DISCONNECT_MESSAGE)
```

Comment:
This is raised when `receive()` is called on the connection object after we have already received a `websocket.disconnect` message. Should we raise `WebSocketDisconnect` as I've done here (which is consistent with the unreachable code that I've removed in `receive_data()`), or leave it as is and continue to raise the base exception?

Reply:
Since `WebSocketDisconnect` is a subclass of `WebSocketException` I think its okay?

Reply:
It's OK for backward compat - but is is correct - i.e., is calling `receive()` on an already closed disconnected connection a different error? Maybe I'm over thinking.

Reply:
I think it makes sense to raise a `WebSocketDisconnect` here. This way the users can handle disconnection related errors differently from the more generic `WebSocketException` if needed. If the `WebSocketException` is raised, then they would have to check the error code to determine if it was a disconnection or not.

Reply:
OK, will roll with it!